presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/IntegerStatisticsBuilder.java

```diff
@@ -44,14 +54,23 @@ private void addIntegerStatistics(long valueCount, IntegerStatistics value)
         nonNullValueCount += valueCount;
         minimum = Math.min(value.getMin(), minimum);
         maximum = Math.max(value.getMax(), maximum);
+
+        if (value.getSum() != null) {
```

Comment:
I would only execute this try if `overflow` is not already set. Given the way this code is currently written, I'm concerned about repeated throws in a tight loop.

Reply:
There is another problem here.  If the input "value" does not have a sum, then we can not output a sum.